library/std/src/os/unix/net/ancillary.rs

```diff
@@ -583,3 +632,40 @@ impl<'a> SocketAncillary<'a> {
         self.truncated = false;
     }
 }
+
+/// An error returned when trying to add anciallary data that exceeds the buffer capacity.
+#[cfg(any(doc, target_os = "android", target_os = "linux",))]
+#[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+pub struct AddAncillaryError {
+    _priv: (),
+}
+
+impl AddAncillaryError {
+    fn new() -> Self {
+        Self { _priv: () }
+    }
+}
+
+#[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+impl fmt::Debug for AddAncillaryError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("AddAncillaryError").finish()
+    }
+}
+
+#[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+impl fmt::Display for AddAncillaryError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "could not add data to anciallary buffer")
+    }
+}
+
+#[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+impl crate::error::Error for AddAncillaryError {}
+
+#[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+impl From<AddAncillaryError> for io::Error {
```

Comment:
I added this `impl` because it is likely that `add_fds()` and `add_creds()` are going to be used from functions that already return an `io::Result`, so this allows you to just use the `?` operator.

Reply:
- io::Error::new does a lot of allocations. This should probably use io::Error::new_const.
- This should now be `ErrorKind::Uncategorized` instead of `Other`.